### ISSUE TITLE
[emscripten] SDL_WindowID=0 when relasing mouse button outside the canvas

### ISSUE THREAD

**Issue by Diyou** [libsdl-org/SDL#10033]
Pressing any mouse button inside a canvas and releasing it outside sets SDL_WindowID=0 or to an ID of the canvas under the cursor.
This makes it impossible to make a reliable reverse lookup with SDL_GetWindowFromID.

**Comment by icculus** [comment]
This happens because of this code in SDL_PrivateSendMouseButton:

```c
event.button.windowID = mouse->focus ? mouse->focus->id : 0;
```

No mouse focus when it leaves the window.

I'm assuming other platforms are getting around this due to autocapture when dragging outside the window...?

We could probably fake this in our emscripten backend, or at least document it as a quirk of the platform (can you even _have_ multiple SDL windows on Emscripten?)

**Comment by Diyou** [comment]
> can you even _have_ multiple SDL windows on Emscripten?

I'd love if there were a more straight forward solution. In my project to create multiple `SDL_Window`s with webgpu contexts I'm moving the `#canvas` ID to another element to attach the event handlers and I'd only need to account for this quirk to make it work.

**Comment by icculus** [comment]
That's hardcore.  :)

We're not going to get to multiple windows working in 3.2.0, but I guess if it's a matter of just having separate canvas elements for each, we could probably make it work _eventually_...but I suspect there are a lot of assumptions to clean out in the code before that could work.

For the issue at hand, I think we can probably just track that the mouse-up event had a paired mouse-down and report the correct window id.

**Comment by slouken** [comment]
Yeah, other platforms autocapture on mouse down by default. There's a hint for this: SDL_HINT_MOUSE_AUTO_CAPTURE
